### PR TITLE
Update Kubernetes and Helm 3 version support in README.md

### PR DESCRIPTION
**Pull request by jvoravong**
Description:
Update Kubernetes and Helm 3 version support in README.md to align with https://docs.splunk.com/observability/en/gdi/opentelemetry/install-k8s.html#supported-kubernetes-distributions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,16 +88,15 @@ This distribution currently supports:
 
 ## Supported Kubernetes distributions
 
-This helm chart is tested and works with default configurations on the following
-Kubernetes distributions:
+The Helm chart works with default configurations of the main Kubernetes distributions. Use actively supported versions:
```
Comment (rmfitzpatrick) [inline README.md:91]: >Use actively supported versions

Can we add some clarity around this like "support is only provided for <insert provider names> maintained versions per their ~license agreements" and "support for expired versions is best effort and backward incompatible feature adoption cannot be circumvented" (cc @aunshc)

File: README.md [hunk 2]
```diff
@@ -88,16 +88,15 @@ This distribution currently supports:
...
 
-- [Vanilla (unmodified version) Kubernetes](https://kubernetes.io)
-- [Amazon Elastic Kubernetes Service](https://aws.amazon.com/eks)
+- [Vanilla (unmodified version) Kubernetes](https://endoflife.date/kubernetes)
```
Comment (rmfitzpatrick) [inline README.md:93]: endoflife.date is a great service, but I'd think linking to primary sources is preferable.

File: README.md [hunk 3]
```diff
@@ -109,7 +108,9 @@ require additional configurations applied to
 
 The following prerequisites are required to use the helm chart:
 
-- [Helm 3](https://helm.sh/docs/intro/install/) (Helm 2 is not supported)
+- [Helm 3](https://helm.sh/docs/intro/install/)
+  - Helm 3 version support may vary depending on the [supported Kubernetes version](#supported-kubernetes-distributions) you are using and the [Helm supported Kubernetes versions skew](https://helm.sh/docs/topics/version_skew/#supported-version-skew).
```
Comment (rmfitzpatrick) [inline README.md:112]: Would be helpful to state we use 3.11 in maintaining this project for some frame of reference.

File: README.md (hunk 2 above)
Comment (jvoravong) [reply README.md:93]: I recommend keeping the "End of Life" section as it provides valuable links to the primary source at the beginning of each page. Using "End of Life" also standardizes the format for presenting version support across various distributions, making it easier to understand. Parsing the version support documentation for each distribution at the source can be overwhelming. Open to other input.

File: README.md (hunk 3 above)
Comment (jvoravong) [reply README.md:112]: I do want this, but I want it to update the value automatically as needed. I'll see if we can make this a badge.

Comment (jvoravong) [reply README.md:112]: Only added a static reference for now. Will try to add automation later for this in another ticket.

File: README.md (hunk 1 above)
Comment (jvoravong) [reply README.md:91]: I'll propose we iterate on updating this in future PRs and in customer docs and move forward with this current version for this PR.

File: README.md (hunk 2 above)
Comment (jvoravong) [reply README.md]: Done

**Review by atoulme — APPROVED** [review]

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**